languages/python/sqlalchemy-oso/sqlalchemy_oso/compat.py

```diff
@@ -17,7 +17,15 @@ def iterate_model_classes(base_or_registry):
     try:  # 1.3 declarative base.
         # TODO (dhatch): Not sure this is legit b/c it uses an internal interface?
         models = base_or_registry._decl_class_registry.items()
-        yield from {model for name, model in models if name != "_sa_module_registry"}
+        for name, model in models:
+            if name != "_sa_module_registry":
+                if isinstance(
+                    model, sqlalchemy.ext.declarative.clsregistry._MultipleClassMarker
```

Comment:
One of the main changes. My understanding is that we're using SQLAlchemy's builtin module registry (it's what allows you to write `ForeignKey("User")` or whatever the syntax is.

If multiple models have the same classname, they're stored inside this multiple class marker, so that SQLAlchemy can tell you to use fully qualified names (`admins.User`)